app/src/main/java/com/kks/nimblesurveyjetpackcompose/repo/survey/SurveyRepoImpl.kt

```diff
@@ -12,12 +17,31 @@ import kotlinx.coroutines.flow.flow
 import javax.inject.Inject
 
 class SurveyRepoImpl @Inject constructor(@ServiceQualifier private val api: Api) : SurveyRepo {
-    override fun getSurveyDetails(surveyId: String): Flow<ResourceState<List<IncludedResponse>>> =
+    override fun getSurveyDetails(surveyId: String): Flow<ResourceState<List<SurveyQuestion>>> =
         flow {
             emit(ResourceState.Loading)
             val apiResult = safeApiCall(Dispatchers.IO) { api.getSurveyDetail(surveyId = surveyId) }
             when (apiResult) {
-                is ResourceState.Success -> emit(ResourceState.Success(apiResult.data.included.orEmpty()))
+                is ResourceState.Success -> {
+                    apiResult.data.included?.let { includedList ->
+                        val answers = includedList.filterIsInstance<IncludedAnswerResponse>()
+                            .map { it.toSurveyAnswer() }
+                            .groupBy { it.id }
+                        val questions = includedList.filterIsInstance<IncludedQuestionResponse>()
+                            .map { questionResponse ->
+                                questionResponse.toSurveyQuestion().also { surveyQuestion ->
+                                    val tempAnsList = arrayListOf<SurveyAnswer>()
+                                    questionResponse.relationships?.answers?.data?.forEach { surveyDataResponse ->
+                                        answers[surveyDataResponse.id]?.first { surveyAnswer ->
+                                            tempAnsList.add(surveyAnswer)
+                                        }
+                                    }
+                                    surveyQuestion.answers = tempAnsList
```

Comment:
This is what we're doing here, right? 🤔 
```suggestion
                                    surveyQuestion.answers = questionResponse.relationships?.answers?.data
                                        ?.mapNotNull { answers[it.id]!!.firstOrNull() }
                                        ?: emptyList()
```